kernel/src/main/java/org/kframework/compile/AddImplicitComputationCell.java

```diff
@@ -51,10 +50,10 @@ public Sentence apply(Module m, Sentence s) {
 
         if (s instanceof RuleOrClaim) {
             RuleOrClaim rule = (RuleOrClaim) s;
-            return rule.newInstance(apply(rule.body(), m), rule.requires(), rule.ensures(), rule.att());
+            return rule.newInstance(apply(rule.body(), m, rule instanceof Claim), rule.requires(), rule.ensures(), rule.att());
```

Comment:
Nitpicking here.
If we are to differentiate between Rule and Claim, why not do it at the top level?
Have two different `apply` functions instead.